### PR TITLE
add sparse constructors for special matrices

### PR DESCRIPTION
**Pull request by bnels**
Added sparse constructors for SymTridiagonal, Tridiagonal, and Bidiagonal matrices
Added tests to sparse test file
Tests work.

### CODE REVIEW

File: base/sparse/sparsematrix.jl [hunk 1]
```diff
@@ -309,6 +309,28 @@ sparse(I,J,V::AbstractVector{Bool},m,n) = sparse(I, J, V, Int(m), Int(n), OrFun(
 
 sparse(I,J,v::Number,m,n,combine::Union(Function,Func)) = sparse(I, J, fill(v,length(I)), Int(m), Int(n), combine)
 
+function sparse(T::SymTridiagonal)
+	m = length(T.dv)
```
Comment (tkelman) [inline base/sparse/sparsematrix.jl]: please use spaces, not tabs

File: base/sparse/sparsematrix.jl [hunk 2]
```diff
@@ -309,6 +309,28 @@ sparse(I,J,V::AbstractVector{Bool},m,n) = sparse(I, J, V, Int(m), Int(n), OrFun(
...
+	return sparse([1:m;2:m;1:m-1],[1:m;1:m-1;2:m],[T.d;T.dl;T.du], Int(m), Int(m))
+end
+
+function sparse(B::Bidiagonal)
+	m = length(B.dv)
+	B.isupper || return sparse([1:m;2:m],[1:m;1:m-1],[B.dv;B.ev], Int(m), Int(m)) # lower bidiagonal
+	return sparse([1:m;1:m-1],[1:m;2:m],[B.dv;B.ev], Int(m), Int(m)) # upper bidiagonal
+end
+
+function find(S::SparseMatrixCSC)
```
Comment (tkelman) [inline base/sparse/sparsematrix.jl]: this is identical to the function below, why is it duplicated?

**Comment by bnels** [comment]
Fixes issue #10837

**Comment by andreasnoack** [comment]
@bnels Thanks

**Comment by tkelman** [comment]
This should really not use the `sparse` constructor if it doesn't have to, it would be considerably faster to construct `colptr`, `rowval`, and `nzval` in CSC sorted order directly since that's straightforward to do for these cases.

**Comment by ViralBShah** [comment]
That will certainly be faster.

**Comment by ViralBShah** [comment]
It may be useful to have some helper functions to help create/modify CSC data structures that do some error checking and such.

**Pull request merged**